### ISSUE TITLE
Incorrect CS8618 warning for generic property with Enum constraint

### ISSUE THREAD

**Issue by mogikanin** [dotnet/roslyn#64040]
**Version Used**: 
VisualStudio.17.Release/17.3.3+32825.248

**Steps to Reproduce**:

![image](https://user-images.githubusercontent.com/840681/190321944-7395885f-141a-4c1d-97b9-e696d9a1673a.png)

**Comment by Youssef1313** [comment]
The warning is correct. `MessageType` could be null, while it shouldn't.

```csharp
https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK7wCYgNQB8DEMqANsRMMQKYAElM5VAsAFAsACATAIwssBuEZNQAe1ALzUYlAO7UAwgB4AokTAA+ABQBKANy9WzNgGZqneQoAqa6tIAWlZDQvUQ1FajAsA3i2p/TJs4Iesz+AfLavv4+oWH+UABm1BoI1FAAzpIkxFpRcdQx+fkItsgA9rJSskrCAMaUAA4IUGUw2iFF1AC+ed0sXUA===
```

What you need is `where T : struct, Enum`

**Comment by mogikanin** [comment]
@Youssef1313 yes, you are right. Forgot about this additional `struct` constraint.
